test/date_time_parser_test.rb

```diff
@@ -53,4 +53,13 @@
     @date_parser.parse(date).must_equal date
     @time_parser.parse(time).must_equal time
   end
+
+  it "raises ArgumentError when parsing a bad string" do
+    proc { @date_parser.parse("asdf") }.must_raise ArgumentError
```

Comment:
I was initially testing the exception message, but the problem is that different ruby versions give different messages. Using `Date.parse` or ` DateTime.strptime` will always just give "ArgumentError: invalid date". On ruby 2.3, `Time.zone.local` ends up giving "ArgumentError: argument out of range", but on 2.4 it can give "ArgumentError: mon out of range" or "ArgumentError: mday out of range" as well as "ArgumentError: argument out of range".